src/searchModal.js

```diff
@@ -158,6 +161,26 @@ export default function searchModalFactory(config) {
         });
     }
 
+    /**
+     * Loops through each class in received class tree and sets access mode to 'denied' on private classes
+     * so are disabled on class tree
+     * @param {object} classTree - class tree received by server, containing resources (classes) and permissions
+     */
+    function manageClassTreePermissions(classTree) {
+        const disableBlockedClasses = function (resources) {
+            resources.forEach((resource, index, array) => {
```

Comment:
**though:** Are you sure the `resources` variable is always an array?

Reply:
Yes, for what I have seen, [this is what is being used](https://github.com/oat-sa/tao-core/blob/master/models/classes/resources/ResourceService.php#L164) in the BE, which is called [from here](https://github.com/oat-sa/tao-core/blob/ea9abe98edbddc7d75df0c212dfb68c3b00b32e6/actions/class.RestResource.php#L147), which is the endpoint of FE request.

Reply:
Well, ok. But the reason why of my question is there no distinction in PHP between pure array and associative array. So that a PHP array can be either translated into pure array or into an object when it comes to JSON, depending if the keys are numeric and contiguous or not.